src/ray/rpc/grpc_client.h

```diff
@@ -96,9 +96,26 @@ class GrpcClient {
   ray::Status CallMethod(
       const PrepareAsyncFunction<GrpcService, Request, Reply> prepare_async_function,
       const Request &request, const ClientCallback<Reply> &callback) {
-    auto call = client_call_manager_.CreateCall<GrpcService, Request, Reply>(
-        *stub_, prepare_async_function, request, callback);
-    return call->GetStatus();
+    uint8_t retries = 0;
+    grpc::Status status;
+    for (uint8_t retries = 0; retries < 5; ++retries) {
+      status = client_call_manager_
+                   .CreateCall<GrpcService, Request, Reply>(
+                       *stub_, prepare_async_function, request, callback)
+                   ->GetStatus();
+      // Retry requests that failed with a transient error.
+      // https://grpc.github.io/grpc/core/md_doc_statuscodes.html.
+      if (status.error_code() == grpc::UNAVAILABLE) {
```

Comment:
Do we also need to check the status in the callback?

Reply:
Good question. It seems that we only get `grpc::UNAVAILABLE` in the callback when the server starts processing the RPC and then crashes partway through, so I don't think we want to check it.